src/transforms.h

```diff
@@ -64,27 +77,33 @@ class ScTransformer : public scBaseVisitor
         }
 
 
-        std::shared_ptr<FunctionDefinition<LatticeType>> functionDef;
+        std::shared_ptr<PropagationNode<LatticeType>> functionDef;
         std::shared_ptr<FunctionEntryNode<LatticeType>> entry;
         if (ctx->type() != nullptr)
         {
-            functionDef = std::make_shared<FunctionDefinition<LatticeType>>(ctx->ID()->getText(), parameters, ctx->type()->getText());
-            entry = create_entry_node(functionDef);
+            std::string syntax = ctx->type()->getText() + " " + ctx->ID()->getText();
+            syntax += stringify_parameters(parameters);
+
+            functionDef = std::make_shared<PropagationNode<LatticeType>>(syntax);
+            entry = create_entry_node(functionDef, parameters, ctx->ID()->getText());
```

Comment:
This entry node is never linked?

Reply:
okay, It's actually linked when adding the return expression. Can we move the linking of the entry node to the bottom of the function? To make sure it always links the entry node

Reply:
The `create_entry_node` function links the entry node